stablehlo/transforms/StablehloProbe.cpp

```diff
@@ -0,0 +1,100 @@
+/* Copyright 2023 The StableHLO Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include <string>
+
+#include "mlir/IR/BuiltinAttributes.h"
+#include "mlir/IR/BuiltinOps.h"
+#include "mlir/IR/Value.h"
+#include "mlir/Support/LLVM.h"
+#include "stablehlo/dialect/StablehloOps.h"
+#include "stablehlo/reference/InterpreterOps.h"
+#include "stablehlo/transforms/Passes.h"
+
+namespace mlir {
+namespace stablehlo {
+
+#define GEN_PASS_DEF_STABLEHLOPROBEPASS
+#include "stablehlo/transforms/Passes.h.inc"
+
+namespace {
+
+class StablehloProbePass
+    : public impl::StablehloProbePassBase<StablehloProbePass> {
+ public:
+  explicit StablehloProbePass() = default;
+  void runOnOperation() override;
+
+ private:
+  // Create a uniquely identifying probe ID for the given value being
+  // instrumented. Attempt to use named MLIR location data, otherwise use an
+  // increasing ID.
```

Comment:
Be a little more specific on how debuginfo is used. Looks like it is before `@` ?